.github/workflows/release.yaml

```diff
@@ -9,14 +9,6 @@ permissions:
   contents: write
 
 jobs:
-  go-version:
-    runs-on: ubuntu-latest
-    outputs:
-      version: ${{ steps.go-version.outputs.version }}
-    steps:
-      - uses: actions/checkout@v2
-      - id: go-version
-        run: echo "::set-output name=version::$(cat ./.go-version)"
```

Comment:
Are we required to remove the `actions/checkout` job? I thought we needed this is in our workflow to access the repo under `$GITHUB_WORKSPACES`

Reply:
I saw that this is replaced with the pinned commit hash for checkout in #1961. Looks good!

Reply:
In #1961 I made bulk blind updates for all workflow files. :)

This one is more specific to the latest changes in the action. `go-version` is one of the jobs that only reads the Golang version and pass the result to jobs down the road. With the latest version of the action `hashicorp/ghaction-terraform-provider-release` we can rid of this step. Once it is done we should be able to rid of `.go-version` file and rely on `go.mod` in all out actions.